packages/blade/src/components/Input/BaseInput/useTaggedInput.ts

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+import type { BaseInputProps } from './BaseInput';
+import type { BladeElementRefWithValue } from '~utils/types';
+import type { FormInputOnEvent, FormInputOnKeyDownEvent } from '~components/Form/FormTypes';
+// import { isReactNative } from '~utils';
+import { getTagsGroup } from '~components/Tag/getTagsGroup';
+import { isReactNative } from '~utils';
+
+type TaggedInputProps = {
+  isTaggedInput?: boolean;
+  tags?: string[];
+  onTagChange?: ({ tags }: { tags: string[] }) => void;
+};
+
+type UseTaggedInputProps = TaggedInputProps &
+  Pick<BaseInputProps, 'isDisabled' | 'onChange' | 'name' | 'value' | 'defaultValue'> & {
+    inputRef: React.RefObject<BladeElementRefWithValue | null>;
+  };
+
+const useTaggedInput = ({
+  tags,
+  isDisabled,
+  onTagChange,
+  isTaggedInput,
+  inputRef,
+  onChange,
+  name,
+  value,
+  defaultValue,
+}: UseTaggedInputProps): {
+  activeTagIndex: number;
+  setActiveTagIndex: (activeTagIndex: number) => void;
+  getTags: ({ size }: { size: NonNullable<BaseInputProps['size']> }) => React.ReactElement[];
+  handleTaggedInputKeydown: (e: FormInputOnKeyDownEvent) => void;
+  handleTaggedInputChange: FormInputOnEvent;
+  handleTagsClear: () => void;
+} => {
```

Comment:
Factor out this type. Looks a bit unreadable inlined like this. 

type UseTaggedInputReturn = {}